spec/services/proofing/lexis_nexis/request_signer_spec.rb

```diff
@@ -0,0 +1,44 @@
+require 'rails_helper'
+
+describe Proofing::LexisNexis::RequestSigner do
+  let(:message_body) { 'APPLICANT_DATA' }
+  let(:path) { '/request/path' }
+  let(:ts) { Time.zone.now.strftime('%s%L') }
+  let(:nonce) { SecureRandom.uuid }
+  let(:config) do
+    OpenStruct.new(
+      base_url: 'https://example.gov',
+      hmac_key_id: 'KEY_ID',
+      hmac_secret_key: 'SECRET_KEY',
+    )
+  end
+
+  subject do
+    Proofing::LexisNexis::RequestSigner.new(
+      config: config,
+      message_body: message_body,
+      path: path,
+    )
+  end
+
+  describe 'generating a valid hmac authorization' do
+    it 'succeeds' do
+      authorization = subject.hmac_authorization(ts: ts, nonce: nonce)
+      regex = %r{
+        HMAC-SHA256\s
+        keyid=#{config.hmac_key_id},\s
+        ts=#{ts},\s
+        nonce=#{nonce},\s
+        bodyHash=(?<hmac>.*),\s
+        signature=(.*)
```

Comment:
I'd recommend `\S` in the regexes so they capture any non-space character, and `+` so they require at least one?
```suggestion
        bodyHash=(?<hmac>\S+),\s
        signature=(\S+)
```